app-frontend/src/app/core/services/project.service.js

```diff
@@ -189,6 +192,38 @@ export default (app) => {
         updateProject(params) {
             return this.Project.updateProject(params).$promise;
         }
+
+        getProjectLayerURL(project, token) {
+            let host = this.$location.host();
+            let protocol = this.$location.protocol();
+            let port = this.$location.port();
+            let formattedPort = port !== 80 && port !== 443 ? ':' + port : '';
+            let userId = 'rf_airflow-user';
+
+            let params = {
+                tag: new Date().getTime()
+            };
+
+            if (token) {
+                params.token = token;
+            }
+
+            let formattedParams = L.Util.getParamString(params);
+
+            return `${protocol}://${host}${formattedPort}` +
+                `/tiles/${project.organizationId}/${userId}` +
+                `/project/${project.id}/{z}/{x}/{y}/${formattedParams}`;
+        }
+
+        getProjectShareURL(project) {
+            let host = this.$location.host();
+            let protocol = this.$location.protocol();
+            let port = this.$location.port();
+            let formattedPort = port !== 80 && port !== 443 ? ':' + port : '';
+
+            return `${protocol}://${host}${formattedPort}` +
+                `/share/${project.id}`;
```

Comment:
Yeah, I also suspect we may have to do some routing rules at the nginx level to properly handle urls without hashes